pub/README.md

```diff
@@ -0,0 +1,188 @@
+## `dependabot-pub`
+
+Dart (pub) support for [`dependabot-core`][core-repo].
+
+### Limitations
+
+ - No support for updating git-dependencies
+   * `dart pub` in general doesn't read versions from git, so upgrade logic is limited.
+   * Some variant of support for updating git-dependencies could be added in the future.
+ - No support for security advisory integration.
+ - If the version found is ignored (by dependabot config) no update will happen (even if, an earlier version could be used)
```

Comment:
This is the right behavior. We don't expect Dependabot to ever downgrade a dependency.

Reply:
This is not about downgrading. The scenario is as follows:
Suppose we have 3 versions of a dependency `dep`:

* dep 1.0.0 <- current
* dep 1.5.0 <- available
* dep 1.7.0 <- ignored by dependabot

One could hope dependabot would suggest upgrading to 1.5.0 as it is not ignored. But because `dependency_services` doesn't know about the ignored versions it will suggest 1.7.0, and that will get filtered away by this implementation. So by ignoring 1.7.0 you also don't get 1.5.0.

Reply:
Something I was wondering, would it make sense for `dependency_services` to know about the ignored versions? For the initial beta release I think this would be fine, but it'd be nice if we could push the ignored versions into pub and have it resolve things using that.

This would also go some way towards dependabot not having to worry about `pub` requirements, since it could just forward the `ignored_versions` as strings.

I can also imagine this makes sense for end users when updating a package via the CLI, I've definitely been in situations where I _know_ version x of some dependency is not compatible for whatever reason, and if I can tell the package manager "update my dependency but not to 1.7.0" that would be nice.

Alternatively,  could `dependency_services` return a list of possible versions to update to, which dependabot could then filter to exclude any ignored versions?

Not expecting this to be fixed right now, just riffing on possible solutions for the future.

Reply:
Yes I think it would absolutely make sense.

The pub version solver currently doesn't have a concept of ignored versions, so without changes to the solver itself pub would not be able to use that information thats why we left it out of this initial MVP.

It could certainly be done.